### PR TITLE
Feature/cpp api tests

### PR DESCRIPTION
**Pull request by figueroa1395**
Replace the (now deprecated) C API tests by re-routing them to the C++ API (introduced in #688) tests. Additionally, the dependence on the C++ core is to be removed in the new tests for the C++ API (Such refactoring is in the scope of this PR, and not in #688, where the new tests were initially introduced).

### CODE REVIEW

**Comment by figueroa1395** [comment]
I intended to isolate the tests from #688 here, but it did not work as intended.

File: tests/c_api_tests/CMakeLists.txt [hunk 1]
```diff
@@ -15,6 +18,7 @@ target_link_libraries(power_grid_model_c_api_tests
     PRIVATE
         power_grid_model
```
Comment (TonyXiang8787) [inline tests/c_api_tests/CMakeLists.txt:19]: my proposal is to remove this one.

**Comment by mgovers** [comment]
please extend the description to something more ... well... descriptive 😬

**Comment by figueroa1395** [comment]
Accidentally closed. Reopening.

File: tests/c_api_tests/CMakeLists.txt (hunk 1 above)
Comment (figueroa1395) [reply tests/c_api_tests/CMakeLists.txt]: Done.

**Review by mgovers — APPROVED** [review]

**Pull request merged**